### PR TITLE
Correctly handle dark icons in `data-icon` and `data-icon-disabled`

### LINKED ISSUE #6869: Wrong images/icons on darkmode on publish/unpublish
### Affected version(s)

5.2.x, 5.3.x

### Description

When publishing / unpublishing any item, the images are not properly loaded. It will use the lightmode images.

See:
image

### PR DESCRIPTION
**Pull request by zoglo**
Fixes #6869

___

**Description**

I extract the incoming data-attributes and properly distribute them for the two states now, this has been missing before.
Attributes get merged again but I remove the data-icons from the HtmlAttributes before doing so.

___

**Edit**
- Just replacing the icon sources within darkAttributes works

### CODE REVIEW

File: core-bundle/contao/library/Contao/Image.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public static function getPath($src)
 
 		$src = rawurldecode($src);
 
-		if (strpos($src, '/') !== false)
+		if (str_contains($src, '/'))
```
Comment (zoglo) [inline core-bundle/contao/library/Contao/Image.php]: That was the cs-fixer - Seems to be the same in 5.3 and 5.x already

Comment (fritzmg) [reply core-bundle/contao/library/Contao/Image.php]: Your dependencies were probably on the `5.3` or `5.x` state. May be you need to run `composer update && composer bin all install` again, before running the CS fixer etc.

Comment (zoglo) [reply core-bundle/contao/library/Contao/Image.php]: That did work @fritzmg

**Comment by leofeyer** [comment]
Yes please.

**Comment by leofeyer** [comment]
Would it not be much shorter if we just added the following in line 210?

```php
foreach (array('data-icon', 'data-icon-disabled') as $icon)
{
	if (isset($darkAttributes[$icon]))
	{
		$pathinfo = pathinfo($darkAttributes[$icon]);
		$darkAttributes[$icon] = $pathinfo['filename'] . '--dark.' . $pathinfo['extension'];
	}
}
```

https://github.com/contao/contao/blob/e2ea04c1f46d1154bd0e1ddfd791df442016c67e/core-bundle/contao/library/Contao/Image.php#L207-L209

**Comment by zoglo** [comment]
leofeyer ~~it's ready to review, just wanted to make sure and test it once again~~

**Comment by zoglo** [comment]
> Would it not be much shorter if we just added the following in line 210?

Guess after trying out multiple approaches, I didn't see this. That works as well @leofeyer, you are right.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @zoglo.

**Pull request merged**